### PR TITLE
refactor(mcp): collapse coerce-list field annotations behind type aliases

### PR DESCRIPTION
**Pull request by dougborg**
## Summary

Polish PR for the list-coercion feature shipped in `mcp-v0.44.1` (commit `30f3fd86`). The original work landed straight on main without review due to a `git push` mishap; we ran three review agents on the commit ourselves to claw back the review value, and this PR addresses their findings.

## Findings addressed

**Reuse — collapse repeated boilerplate.** The phrase `Annotated[list[int] | None, BeforeValidator(coerce_str_list_input)]` repeated verbatim 11 times across 6 tool files. Adds six prebuilt aliases in `list_coercion.py`:

- `CoercedStrList` / `CoercedStrListOpt`
- `CoercedIntList` / `CoercedIntListOpt`
- `CoercedStrIntList` / `CoercedStrIntListOpt`

Refactors the 11 sites to use the appropriate alias. Each per-field declaration goes from three lines of type-noise to a single token:

```diff
- ids: Annotated[list[int] | None, BeforeValidator(coerce_str_list_input)] = Field(
-     default=None,
-     description="Filter by explicit list of purchase order IDs. ..."
- )
+ ids: CoercedIntListOpt = Field(
+     default=None,
+     description="Filter by explicit list of purchase order IDs. ..."
+ )
```

**Quality — uncovered edge cases.** Adds four tests:

- Mixed-type list passthrough (`["WS74001", 12345]` for `list[str | int]`)
- CSV with numeric items for `list[str | int]` (left-side `str` of the union wins)
- Empty string + `min_length=1` interaction → raises `too_short`, not `list_type` (documents the UX so a future contributor can't accidentally regress)
- Non-numeric string for `list[int]` → per-item `int_parsing` error (useful diagnostic)

**Efficiency — clean.** No changes warranted; the validator already short-circuits non-strings, gates JSON parsing on a leading `[`, and pydantic builds the schema once at import time.

## Files changed

- `katana_mcp_server/src/katana_mcp/tools/list_coercion.py` — adds six type aliases (~10 LoC)
- 6 tool files (`items.py`, `inventory.py`, `manufacturing_orders.py`, `purchase_orders.py`, `sales_orders.py`, `reporting.py`) — replace verbose `Annotated[...]` declarations with the new aliases
- `katana_mcp_server/tests/tools/test_list_coercion.py` — adds 4 tests; updates the test stub to use the aliases (still 19 tests total, all passing)
- `uv.lock` — reflects 0.44.0 → 0.44.1 from the prior release commit

## Test plan

- [x] `uv run poe check` — all 2,509 tests pass (4 new vs. 2,505 in v0.44.1)
- [ ] Verify behavior unchanged: any existing call site that worked at v0.44.1 still works after this refactor (pure rename + alias)

## Branch-protection note

The reason this PR exists is that `30f3fd86` was force-pushed straight to main when `git push -u origin <branch>` resolved its tracked upstream as `main`. We've added a memory note about always using `git push -u origin HEAD:refs/heads/<name>` for first-time pushes — but the cleaner fix is **branch protection on main** that requires PRs + status checks. Worth adding as a follow-up.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: katana_mcp_server/tests/tools/test_list_coercion.py [hunk 1]
```diff
@@ -14,22 +14,23 @@
...
-    skus: Annotated[list[str] | None, BeforeValidator(coerce_str_list_input)] = Field(
-        default=None
-    )
-    ids: Annotated[list[int] | None, BeforeValidator(coerce_str_list_input)] = Field(
-        default=None
-    )
+    skus: CoercedStrListOpt = Field(default=None)
+    ids: CoercedIntListOpt = Field(default=None)
+    # Mirrors check_inventory's required + min_length=1 shape so we can
+    # test the empty-string + min_length interaction in one place.
```
[the remark this answers is not in the document]
Comment (dougborg) [reply katana_mcp_server/tests/tools/test_list_coercion.py]: Good catch — the comment was misleading on two counts: it claimed `min_length=1` parity with `check_inventory` while the field is declared `min_length=0` (a no-op), and it claimed this field is what tests the empty-string + `min_length` interaction when actually that test (`test_empty_string_plus_min_length_one_raises_min_length_error`) uses an inline model declared inside the test itself.

Fixed in 26740c62 — rewrote the comment to describe what `required_mixed` actually is (a stub of the `CoercedStrIntList` alias for the mixed-type tests below, with `default_factory` so `_build()` can omit it) and points at the inline test that covers the empty-string interaction. Also dropped the redundant `min_length=0`.
